pkg/gardenlet/controller/shoot/shoot_control_migrate.go

```diff
@@ -126,7 +126,8 @@ func (c *Controller) runPrepareShootControlPlaneMigration(o *operation.Operation
 			return nil
 		}),
 		utilerrors.ToExecute("Retrieve the Shoot namespace in the Seed cluster", func() error {
-			if err := botanist.K8sSeedClient.Client().Get(ctx, client.ObjectKey{Name: o.Shoot.SeedNamespace}, namespace); err != nil {
+			botanist.SeedNamespaceObject = &corev1.Namespace{}
+			if err := botanist.K8sSeedClient.Client().Get(ctx, client.ObjectKey{Name: o.Shoot.SeedNamespace}, botanist.SeedNamespaceObject); err != nil {
```

Comment:
Isn't the better approach to add `.DoIf(nonTerminatingNamespace)` to the the task that causes the panic (`botanist.WakeUpKubeAPIServer` in this case)?

Wouldn't the following fix the issue?

```diff
diff --git a/pkg/gardenlet/controller/shoot/shoot_control_migrate.go b/pkg/gardenlet/controller/shoot/shoot_control_migrate.go
index 06d06dbc9..b0ac9e6fb 100644
--- a/pkg/gardenlet/controller/shoot/shoot_control_migrate.go
+++ b/pkg/gardenlet/controller/shoot/shoot_control_migrate.go
@@ -183,7 +183,7 @@ func (c *Controller) runPrepareShootControlPlaneMigration(o *operation.Operation
                })
                wakeUpKubeAPIServer = g.Add(flow.Task{
                        Name:         "Scaling Kubernetes API Server up and waiting until ready",
-                       Fn:           flow.TaskFn(botanist.WakeUpKubeAPIServer).DoIf(wakeupRequired),
+                       Fn:           flow.TaskFn(botanist.WakeUpKubeAPIServer).DoIf(wakeupRequired && nonTerminatingNamespace),
                        Dependencies: flow.NewTaskIDs(deployETCD, scaleETCDToOne),
                })
                ensureResourceManagerScaledUp = g.Add(flow.Task{
```

Reply:
`wakeupRequired` is only true if `nonTerminatingNamespace` is already true as well. 
```
cleanupShootResources   = nonTerminatingNamespace && kubeAPIServerDeploymentFound
wakeupRequired          = (o.Shoot.Info.Status.IsHibernated || (!o.Shoot.Info.Status.IsHibernated && o.Shoot.HibernationEnabled)) && cleanupShootResources
```
The nil pointer exception happens always when SNI is enabled and the shoot is hibernated (not only when the namespace is being already deleted)